norman_utils_external/singleton.py

```diff
@@ -1,4 +1,17 @@
 class Singleton(type):
+    """
+    Metaclass implementing the Singleton design pattern.
+
+    Classes using this metaclass will only ever have one instance.
+    Any attempt to instantiate the class again returns the same
+    previously created object.
+
+    **Attributes**
+
+    - **_instances** (`dict[type, Any]`)
+      Internal cache mapping classes to their singleton instances.
```

Comment:
Wording: The internal cache mapping classes to their singleton instances. Ensures no class whose metadata class is the Singleton class will be instantiated twice.